src/http/server/handlers/websocket_handler.cr

```diff
@@ -14,7 +14,7 @@ class HTTP::WebSocketHandler
   end
 
   def call(context)
-    if context.request.headers["Upgrade"]? == "websocket" && context.request.headers.includes_word?("Connection", "Upgrade")
+    if context.request.headers["Upgrade"]?.try(&.compare("websocket", case_insensitive: true)) == 0 && context.request.headers.includes_word?("Connection", "Upgrade")
```

Comment:
This is now a very long line of code, it might be good to refactor out `upgrade_header = context.request.headers["Upgrade"]?` and use `if upgrade_header && upgrade_header.compare("websocket", case_insensitive: true) == 0 && context.request.headers.includes_word?("Connection", "Upgrade")` but that's still rather long.

How about structuring this as a series of guards:
```cr
call_next(context) unless upgrade_header && upgrade_header.compare("websocket", case_insensitive: true) == 0
call_next(context) unless context.request.headers.includes_word?("Connection", "Upgrade")
```